addons/analytic/models/analytic_plan.py

```diff
@@ -0,0 +1,126 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import json
+from odoo import api, fields, models, _
+from random import randint
+
+
+class AccountAnalyticPlan(models.Model):
+    _name = 'account.analytic.plan'
+    _description = 'Analytic Plans'
+    _parent_store = True
+    _rec_name = 'complete_name'
+
+    def _default_color(self):
+        return randint(1, 11)
+
+    name = fields.Char(required=True)
+    description = fields.Text(string='Description')
+    parent_id = fields.Many2one('account.analytic.plan', string="Parent", ondelete='cascade', domain="['|', '|',('id', '!=', id), ('company_id', '=', False), ('company_id', '=', company_id)]")
+    parent_path = fields.Char(index='btree', unaccent=False)
+    children_ids = fields.One2many('account.analytic.plan', 'parent_id', string="Childrens")
+    children_count = fields.Integer('Children Plans Count', compute='_compute_children_count')
+    complete_name = fields.Char('Complete Name', compute='_compute_complete_name', recursive=True, store=True)
+    company_id = fields.Many2one('res.company', string='Company', default=lambda self: self.env.company)
+    account_ids = fields.One2many('account.analytic.account', 'plan_id', string="Accounts")
+    account_count = fields.Integer('Analytic Accounts Count', compute='_compute_analytic_account_count')
+    root_parent_id = fields.Many2one('account.analytic.plan', string="Root Parent", compute="_compute_root_parent",
```

Comment:
root_parent probably not needed, as stored field. 
Might want to use `parent_path.split('/')[0]` instead wherever needed